### PR TITLE
[XLA:CPU][oneDNN] : Add OneDNN softmax rewriter

### PR DESCRIPTION
**Pull request by sachinmuradi**
This PR rewrites the below mentioned HLO pattern, into softmax pattern and executes it through custom call to oneDNN softmax library function.
This pattern is mainly seen in flax/jax [softmax ](https://github.com/google/jax/blob/a259df0d76e50e0e54fa5a69a7c6b78975cde10a/jax/_src/random.py#L1025)operation, with Float32 datatype. The rewrite uses 'divide' operation as root node to match the pattern and is currently called before any layout assignment / float normalization passes to avoid more convert ops or layout normalization ops during pattern matching.

HLO Pattern :
image

### CODE REVIEW

**Comment by sachinmuradi** [comment]
This PR was previously approved here : https://github.com/tensorflow/tensorflow/pull/61589
Later was asked to move to this repo.

**Review by penpornk — APPROVED** [review]
Approving since @d0k already approved the original PR.

**Comment by ddunl** [comment]
I believe conflicts are blocking import, I think rebase is necessary

**Comment by sachinmuradi** [comment]
@ddunl  I rebased my PR, conflicts are now resolved

**Review by ddunl — APPROVED** [review]
Thanks!!

**Comment by sachinmuradi** [comment]
@penpornk Can you help merge this PR ?

**Comment by penpornk** [comment]
@sachinmuradi It's already merged in https://github.com/openxla/xla/commit/ac27e3f77b06ef2b387db5f0eaae2a44fd568e5d. But there seemed to be some infra issues so the changes got mixed with some other changes in the commit. Please let me know if anything is missing.

**Comment by ddunl** [comment]
I think that commit is not yet merged - a comment in this PR caused a subtle Copybara issue, but it should be reflected on main shortly. Sorry about the delay!

**Comment by penpornk** [comment]
Thank you for the update, @ddunl! I'll reopen the PR for now then. We can manually close it later when the commit is actually merged.

**Comment by sachinmuradi** [comment]
@penpornk @ddunl Thank you for taking care of it, let me know if any action needed from my side
